### PR TITLE
Refactor coherence creation script 

### PR DESCRIPTION
**Pull request by jhkennedy**
This is mostly clean-up of some debugging stuff, variable names, and adding type hints.

Also adds:
* back the `environment.yml` which simplifies dev environment setup
* a `.gitignore` for python,vim,pycharm

Drops the intermediate data files. I did upload the France tile list to:
```
s3://ffwilliams2-shenanigans/stac/france_tiles.txt
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Clone the repository, create the environment, and install the developer dependen
 git clone git@github.com:ASFHyP3/asf-stac.git
 cd asf-stac
 
-conda create -c conda-forge -n asf-stac python=3.9 postgresql
+conda env create -f environment.yml
 conda activate asf-stac
 
 make install
```
Comment (jtherrmann) [inline README.md:21]:
```suggestion
```

File: src/coherence_stac.py [hunk 2]
```diff
@@ -66,23 +70,25 @@
     'and is rich in spatial and temporal information for a variety of mapping applications.'
 )
 
+s3 = boto3.client('s3')
```
Comment (forrestfwilliams) [inline src/coherence_stac.py:73]: Are `boto3` resources generally created at the module scope, then not passed to functions? I guess this surprises me.

**Review by forrestfwilliams — APPROVED** [review]
Looks good to me! I left one comment that has a question, but it doesn't affect the code.

File: src/coherence_stac.py (hunk 2 above)
Comment (jhkennedy) [reply src/coherence_stac.py:73]: Sometimes; not always. For scripts like this, where you only even want or need one session (better performance with ice), your can think of the session as a constant, and instantiating it at the module level makes all the interfaces easier.

In a library or anything beyond a "script", probably no. Or at least, you'll take more care with creating and reusing the session

**Pull request merged**